src/main/java/ev3dev/actuators/lego/motors/BaseRegulatedMotor.java

```diff
@@ -206,16 +206,31 @@ public void hold() {
      * Cancels any rotate() orders in progress
      */
     public void stop() {
-		this.setStringAttribute(COMMAND, STOP);
-
-        for (RegulatedMotorListener listener : listenerList) {
-            listener.rotationStopped(this, this.getTachoCount(), this.isStalled(), System.currentTimeMillis());
-        }
+        stop(true);
     }
 
     @Override
-    public void stop(boolean b) {
+    public void stop(boolean immediateReturn) {
+        doStop(HOLD, immediateReturn);

```

Comment:
This is consistent with leJOS:
https://sourceforge.net/p/lejos/ev3/code/ci/master/tree/ev3classes/src/lejos/hardware/motor/BaseRegulatedMotor.java#l196
```
float speed = 0,
int acceleration = <acceleration>,
int limit = <no limit>,
boolean hold = TRUE,
boolean waitComplete = <forwarded>
```